### ISSUE TITLE
Using openExternal() instead of executeCommand("vscode.open") prevents testing

### ISSUE THREAD

**Issue by DanTup** [Microsoft/vscode#69608]
In the latest release there's an `openExternal()` function:

> we have added a new API vscode.env.openExternal. It expects a URL and can be used to open website-links, mail-links, or application-url-handler

However it seems to prevent testing of code that opens URLs. If we use the `vscode.open` command, we can write a test like this:

```ts
test("functionToTest opens browser", function () {
    // Intercept vscode.open so we don't spawn browsers!
    const executeCommand = sinon.stub(vs.commands, "executeCommand").callThrough();
    const open = executeCommand.withArgs("vscode.open", sinon.match.any).resolves();

    // Call the function to test.
    functionToTest();

    // Ensure it attempted to open the browser.
    assert.equal(open.called, true);
});
```

Where `functionToTest()` includes something like:

```ts
vs.commands.executeCommand("vscode.open", vs.Uri.parse("https://www.google.co.uk/"));
```

This works fine. No browser is spawned, and we can verify that the method was called. Given the release notes, I presume that we are encouraged to switch to `openExternal`. However, this is declared as a property and doesn't appear to be stubable (by sinon, at least):

```
     TypeError: Cannot redefine property: openExternal
      at Function.defineProperty (<anonymous>)
      at wrapMethod (node_modules/sinon/lib/sinon/util/core/wrap-method.js:95:16)
      at stub (node_modules/sinon/lib/sinon/stub.js:76:44)
      at Sandbox.stub (node_modules/sinon/lib/sinon/sandbox.js:291:33)
      at Context.<anonymous> (src/test/extension.test.ts:9:15)
```

One option is to just ignore this, and allow the test to spawn a browser (and not be able to confirm it worked). However, if you change the test to just:

```
test("functionToTest opens browser", function () {
    // Call the function to test.
    functionToTest();

    // Test the things we can test (which does not include that a browser spawned)
});
```

And `functionToTest()` to include this:

```
vs.env.openExternal(vs.Uri.parse("https://www.google.co.uk/"));
```

Then when running on Travis Linux (at least) the test run will never complete. The tests complete and the suite `done` event fires, but the VS Code test wrapper never terminates:

```
  Extension Tests
    ✓ functionToTest opens browser
  1 passing (150ms)
Xlib:  extension "RANDR" missing on display ":99.0".
ATTENTION: default value of option force_s3tc_enable overridden by environment.
[6405:6405:0228/173845.767243:ERROR:sandbox_linux.cc(346)] InitializeSandbox() called with multiple threads in process gpu-process.
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
```

Example: https://travis-ci.org/DanTup/vscode-repro-open-external/builds/499931863
Code: https://github.com/DanTup/vscode-repro-open-external

If `openExternal` is the intended way for us to spawn browsers, it should support better testing (especially given how tricky it can be to track down, since it only happens on Travis Linux for me and I didn't have good test results for a weeks there to know what change caused it 😔...)

**Comment by jrieken** [comment]
> If `openExternal` is the intended way for us to spawn browsers, it should support better testing

VS Code is a UI application that's meant to work in an UI environment. We have no plans supporting VS Code without UI.

**Comment by DanTup** [comment]
> VS Code is a UI application that's meant to work in an UI environment. We have no plans supporting VS Code without UI.

I don't understand what this means. I'm not running VS Code without a UI, I'm just trying to run integration tests on a CI system (which works fine) but having any code call that method during a test run results in a hang at the end of the run (the method itself doesn't hang, it returns).

I assumed that `openExternal` was added because you'd prefer we used that to `executeCommand("vs.open")`, however with this issue, it makes no sense to switch (I've switched back). I'm curious of the purpose of this API if it just does the same thing, except it also breaks testing. What are the advantages to using `openExternal`?
